File: Year 2/Semester 4/Programming and Projection Enironments/Week 8/SeventhWeekCSharpServerClient/networking/utils/RpcConcurrentServer.cs
using log4net.Repository.Hierarchy;
using log4net;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using SeventhWeekCSharpServerClient.services;

namespace SeventhWeekCSharpServerClient.networking.utils
{
    public class RpcConcurrentServer : AbsConcurrentServer
    {
        private readonly IServices service;
        private static readonly ILog logger = LogManager.GetLogger("RpcConcurrentServer");
      
        private ClientRpcReflectionWorker worker;
        public RpcConcurrentServer(string host, int port, IServices service) : base(host, port)
        {
            this.service = service; 
            logger.Info("RpcConcurrentServer created");
            Console.WriteLine(@"RpcConcurrentServer created");
        }
        protected override Thread CreateWorker(TcpClient client)
        {
            worker = new ClientRpcReflectionWorker(service, client);
            return new Thread(new ThreadStart(worker.Run));
        }

    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Exam preparation/Ghiceste cuvantul/Ghiceste cuvantul trei jucatori/Networking/src/main/java/martin/network/utils/RpcConcurrentServer.java
package martin.network.utils;

import martin.network.rpcprotocol.ClientRpcReflectionWorker;
import martin.service.IService;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.net.Socket;


public class RpcConcurrentServer extends AbsConcurrentServer {
    private final IService chatServer;
    private static final Logger logger= LogManager.getLogger();
    public RpcConcurrentServer(int port, IService chatServer) {
        super(port);
        this.chatServer = chatServer;
        logger.info("RpcConcurrentServer created");
    }

    @Override
    protected Thread createWorker(Socket client) {
        ClientRpcReflectionWorker worker=new ClientRpcReflectionWorker(chatServer, client);
        Thread tw=new Thread(worker);
        return tw;
    }

    @Override
    public void stop(){
        logger.info("Stopping services ...");
    }
}
